components/feature/pwa/src/main/java/mozilla/components/feature/pwa/ManifestStorage.kt

```diff
@@ -0,0 +1,42 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.pwa
+
+import android.content.Context
+import android.content.Context.MODE_PRIVATE
+import androidx.core.content.edit
+import mozilla.components.concept.engine.manifest.WebAppManifest
+import mozilla.components.concept.engine.manifest.WebAppManifestParser
+
+class ManifestStorage(context: Context) {
+    private val preferences = context.getSharedPreferences(PREFERENCE_NAME, MODE_PRIVATE)
```

Comment:
Instead of using `SharedPreferences` we may want to write manifests to a flat file.
* A SharedPreferences **write** always writes *everything* to disk as XML
* And a SharedPreferences **read** always reads *everything* into memory and will keep it there - even if not referenced.